### PR TITLE
Added dynamic configuration for CoreRestApi on SDK side

### PR DESCRIPTION
**Pull request by dansterenson**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
https://jira-hq.paloaltonetworks.local/browse/CIAC-6112
https://jira-hq.paloaltonetworks.local/browse/CIAC-528

### CODE REVIEW

File: demisto_sdk/commands/test_content/TestContentClasses.py [hunk 1]
```diff
@@ -1263,8 +1263,28 @@ def _set_integration_params(
             self._change_placeholders_to_values(server_url, conf)
             for conf in integration_params
         ]
-
-        if integration_params:
+        if self.name == "Core REST API":
+            if IS_XSIAM:
+                self.configuration.params = {  # type: ignore
```
Comment (dorschw) [inline demisto_sdk/commands/test_content/TestContentClasses.py:1268]: why ignore?

File: demisto_sdk/commands/test_content/TestContentClasses.py [hunk 2]
```diff
@@ -1263,8 +1263,28 @@ def _set_integration_params(
...
+            else:
+                self.configuration.params = {  # type: ignore
+                    "url": "https://localhost",
+                    "creds_apikey": {
+                        "identifier": "",
+                        "password": self.build_context.api_key,
+                    },
+                    "auth_method": "Standard",
+                    "insecure": True,
+                }
```
Comment (dorschw) [inline demisto_sdk/commands/test_content/TestContentClasses.py:1286]: we can avoid code duplication, e.g. `"url": server_url if IS_XSIAM else "https://localhost"` etc

File: demisto_sdk/commands/test_content/TestContentClasses.py [hunk 3]
```diff
@@ -1538,7 +1546,7 @@ def create_integration_instance(
                     if param_conf["display"] in params
                     else param_conf["name"]
                 )
-                if key == "credentials":
+                if key == "credentials" or key == "creds_apikey":
```
Comment (dorschw) [inline demisto_sdk/commands/test_content/TestContentClasses.py:1549]:
```suggestion
                if key in {"credentials", "creds_apikey"}:
```

**Review by dorschw — APPROVED** [review]

File: demisto_sdk/commands/test_content/TestContentClasses.py (hunk 1 above)
Comment (dansterenson) [reply demisto_sdk/commands/test_content/TestContentClasses.py]: self.configuration is of type `Optional[IntegrationConfiguration]` so mypy warns for AttributeError. Through the code `# type: ignore` is used to ignore this warning.

Comment (dansterenson) [reply demisto_sdk/commands/test_content/TestContentClasses.py]: self.configuration is of type `Optional[IntegrationConfiguration]` so mypy warns for AttributeError. Through the code `# type: ignore` is used to ignore this warning.

**Pull request merged**